### PR TITLE
fix(UnstructuredGrid)

### PR DESCRIPTION
**Pull request by cnicol-gwlogic**
Loading an mf6 DISU model with variable voronoi grid results in extraneous iverts entries
of None (Iverts appears to be a consistent (not jagged) shape). This results in a vertexdict
indexing error in UnstructuredGrid._build_grid_geometry_info when using plot() methods.

Filter out None entries from iverts in UnstructuredGrid._build_grid_geometry_info

### CODE REVIEW

**Comment by cnicol-gwlogic** [comment]
This pymake issue is a bit weird. I can replicate locally via a clean install of pymake using pip install git+https://github.com/modflowpy/pymake. This results in only the root directory .py files being downloaded - no subfolders (including ./utils/, which results in the ci error).

**Comment by mwtoews** [comment]
The missing pymake modules should be fixed with https://github.com/modflowpy/pymake/pull/102

**Comment by jdhughes-usgs** [comment]
The pymake issue required a little more work but is fixed now. I reran the GutHub Actions workflow and everything is green now.

**Comment by jlarsen-usgs** [comment]
@cnicol-gwlogic @jdhughes-usgs 
I'll take a look through this. 

From my first glance, it seems like we may have an issue on the either the reader side of the modflow6 code or this is something we could trap for in instantiation of the modelgrid, because there shouldn't be None values supplied to or stored in the `UnstructuredGrid`'s `_vertices` parameter.

**Comment by cnicol-gwlogic** [comment]
@jlarsen-usgs thanks, yep I did wonder. I can set you up a small example if you think it would help. But basically all I did was import a mf6 disu model written by algomesh (so voronoi). Looks like on disu read, the vertices list is sized along the second dimension according to the max number of vertices across all cells.

**Comment by jlarsen-usgs** [comment]
@cnicol-gwlogic 

A small example would definitely help. Thank you! I'll wait for your example problem to investigate this.

**Comment by cnicol-gwlogic** [comment]
@jlarsen-usgs [here ](https://github.com/cnicol-gwlogic/mf6_test_algomesh)we go. See test_mf6_load.ipynb for the error. Thanks a lot. I should turn this into an issue I guess.

**Review by jlarsen-usgs — APPROVED** [review]
I made changes to this PR to filter out None types 1) before being passed to the modelgrid and 2) in the UnstructuredGrid.iverts call, if the user builds the Grid object themselves. 

May need someone else to review this PR now, because I implemented a different approach to this problem than @cnicol-gwlogic

**Comment by langevin-usgs** [comment]
Hey guys, it's unclear to me why there are none values in iverts.  I know we can filter them out, but I'm not sure they should be there in the first place.  @spaulins-usgs, any idea what's going on here with the load of a jagged list?  Either way, I think we should keep the test, and use it to ensure that we load iverts correctly.

**Comment by spaulins-usgs** [comment]
@langevin-usgs  I assume the iverts info is being pulled from the cell2d data (disv package cell2d block).  In the cell2d block icvert is repeating data of variable length.  Flopy for mf6 currently stores this type of data by adding a column to the recarray for each entry of repeating data.  For columns where there is less than the maximum amount of repeating data, None is used to fill in the rest of the column.

**Comment by langevin-usgs** [comment]
Got it.  I see that VertexGrid filters out the None, but UnstructuredGrid does not, so there is an inconsistency that this PR should address.  Does this PR handle this None issue consistently between VertexGrid and UnstructuredGrid implementations?  A quick look suggests the implementations are a little different.

**Comment by jlarsen-usgs** [comment]
@langevin-usgs, @spaulins-usgs  

It looks like for `VertexGrid` we trapped for None types in the cell2d array during the _build_geometry_info step. I personally prefer using the filter() built in on the property method instead of using checking for None while building. Any strong preference on the None filtering method, before I ethier update `UnstructuredGrid` or `VertexGrid` for consistency?

**Comment by langevin-usgs** [comment]
No preference from me. Just thinking about consistency.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Joshua Larsen ***@***.***>
Sent: Tuesday, December 7, 2021 5:29:54 PM
To: modflowpy/flopy ***@***.***>
Cc: Langevin, Christian D ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [modflowpy/flopy] fix(UnstructuredGrid) (PR #1312)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



@langevin-usgs<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flangevin-usgs&data=04%7C01%7Clangevin%40usgs.gov%7C33f649b3671e4f8d772108d9b9d98117%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637745166090252321%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=7CsQuTuqbDN6RycWD%2Fr1MQjSsYCvhpoFN%2B0NldstTc4%3D&reserved=0>, @spaulins-usgs<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fspaulins-usgs&data=04%7C01%7Clangevin%40usgs.gov%7C33f649b3671e4f8d772108d9b9d98117%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637745166090252321%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=OdKb6uLsSdctvnqyVcRavmKgEMVgrVfay78xghkVeE0%3D&reserved=0>

It looks like for VertexGrid we trapped for None types in the cell2d array during the _build_geometry_info step. I personally prefer using the filter() built in on the property method instead of using checking for None while building. Any strong preference on the None filtering method, before I ethier update UnstructuredGrid or VertexGrid for consistency?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmodflowpy%2Fflopy%2Fpull%2F1312%23issuecomment-988339087&data=04%7C01%7Clangevin%40usgs.gov%7C33f649b3671e4f8d772108d9b9d98117%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637745166090262274%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=ObqMfScIU25DwbrvvagOMidgGfs2NMEp%2F0%2B1tEFyvdg%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAZXK46EOLRR5SMJ4MDGOUTUP2KHFANCNFSM5JIHIE6A&data=04%7C01%7Clangevin%40usgs.gov%7C33f649b3671e4f8d772108d9b9d98117%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637745166090262274%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2FRcMkZODvv3hXZi%2FU9%2F7C42fpjqFp63uBHjkHRXlQH0%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Clangevin%40usgs.gov%7C33f649b3671e4f8d772108d9b9d98117%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637745166090272233%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=sXhlJW49yV2aq%2FZ8ldGHnuaxTbJDe5jvJG6rAOOKrDc%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Clangevin%40usgs.gov%7C33f649b3671e4f8d772108d9b9d98117%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637745166090272233%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=fDf8pP3TaF%2BVMtt2YfQav7I9FuwtCYCwixzOjfAeUY0%3D&reserved=0>.

**Comment by langevin-usgs** [comment]
@jlarsen-usgs, I'm noticing that we are getting some weird warnings now based on the filter with None approach implemented here.  See simple example below.  I'm not positive but could this be partially responsible for the failure we are seeing with #1321?  Not sure, but something is fishy here.  Your filter approach seems clever but it is also hard to follow.  

image

**Comment by langevin-usgs** [comment]
After sleeping on it, maybe this is what we want, unless you think the list/filter option is better optimized.

image

**Comment by jlarsen-usgs** [comment]
@langevin-usgs, I think we can swap out the list, filter method for a nested list comprehension without taking much of a hit in optimization.
